src/tools/spin/spinquic.cpp

```diff
@@ -1026,7 +1026,11 @@ void Spin(Gbs& Gb, LockableVector<HQUIC>& Connections, std::vector<HQUIC>* Liste
                 std::lock_guard<std::mutex> Lock(ctx->Lock);
                 auto Stream = ctx->TryGetStream();
                 if (Stream == nullptr) continue;
-                MsQuic.StreamReceiveSetEnabled(Stream, GetRandom(2) == 0);
+                auto StreamCtx = SpinQuicStream::Get(Stream);
+                bool Enable = GetRandom(2) == 0;
+                if (!Enable || StreamCtx->PendingRecvLength == UINT64_MAX) {
```

Comment:
race condition.
This reduce assertion failure occurrence, but RecvBuffer->ReadPending is updated after calling API, then error